apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/parser/sdl/GraphSDLSchemaParser.kt

```diff
@@ -106,6 +109,218 @@ object GraphSDLSchemaParser {
     )
   }
 
+  private fun List<GraphSDLParser.TypeSystemExtensionContext>.parse(
+      typeDefinitions: MutableMap<String, GraphSdlSchema.TypeDefinition>
```

Comment:
not a big fan of passing mutable map and having side affect to mutate `typeDefinitions` map, but don't see easy way to make it pure. We might consider to rename this function to smth that describes mutation maybe like `MutableMap<String, GraphSdlSchema.TypeDefinition>.decorateWithTypeExtentantion()`?

Reply:
Yea, not a huge fan for mutable maps either. I just pushed a new version with a `fold`. That's a lot more copying Maps all over the place but I think that's the price to pay for immutability.